src/webhook-event-check.ts

```diff
@@ -31,9 +32,59 @@ async function webhookEventCheck (app: Application, eventName: string) {
  * @returns Returns `true` when the application is subscribed to a webhook
  * event. Otherwise, returns `false`. Returns `undefined` if Probot failed to
  * retrieve GitHub App metadata.
+ *
+ * **Note**: Probot  will only check against a list of events known to be in the
+ * `/app` response. Therefore, only the `false` value returned should be
+ * considered truthy.
  */
 async function isSubscribedToEvent (app: Application, baseEventName: string) {
-  if (baseEventName === '*') {
+  // A list of events known to be in the response of `/app`. This list can be
+  // retrieved by calling `GET /app` from an authenticated app that has maximum
+  // permissions and is subscribed to all available webhook events.
+  const knownBaseEvents = [
+    'check_run',
+    'check_suite',
+    'commit_comment',
+    'content_reference',
+    'create',
+    'delete',
+    'deployment',
+    'deployment_status',
+    'deploy_key',
+    'fork',
+    'gollum',
+    'issues',
+    'issue_comment',
+    'label',
+    'member',
+    'membership',
+    'milestone',
+    'organization',
+    'org_block',
+    'page_build',
+    'project',
+    'project_card',
+    'project_column',
+    'public',
+    'pull_request',
+    'pull_request_review',
+    'pull_request_review_comment',
+    'push',
+    'release',
+    'repository',
+    'repository_dispatch',
+    'star',
+    'status',
+    'team',
+    'team_add',
+    'watch'
+  ]
+
+  // Because `/app` does not include many events - such as events that all
```

Comment:
> does not include many events

My understanding is that it includes nearly all events? Maybe change the comment to reflect that? We could say something like

```suggestion
  // Because `/app` does not include all events - such as events that all
```

Reply:
6c2fe55fe7017075e690f5882295120ef6cd91bf

Reply:
Your understanding is correct. Default events won't be included in the `GET /app` response. Therefore, we can't check for the event `marketplace_purchase` in the `GET /app` response, even though the app is subscribed to that default event. So what I've done is make the program only check events that could be in `GET /app`. We're not losing much if we can't verify a default event because... well, all apps are subbed to the event anyways.